src/filesystem/impls/appshell/AppshellFileSystem.js

```diff
@@ -363,11 +369,11 @@ define(function (require, exports, module) {
             if (stat.size > (FileUtils.MAX_FILE_SIZE)) {
                 callback(FileSystemError.EXCEEDS_MAX_FILE_SIZE);
             } else {
-                appshell.fs.readFile(path, encoding, function (_err, _data, encoding) {
+                appshell.fs.readFile(path, encoding, function (_err, _data, encoding, preserveBOM) {
                     if (_err) {
                         callback(_mapError(_err));
                     } else {
-                        callback(null, _data, encoding, stat);
+                        callback(null, _data, encoding, preserveBOM, stat);
```

Comment:
Remove the preserveBOM marker from callback. It should be consumed internally and kept in File prototype only.